### PR TITLE
[#538] Wire websocket server to the redux store

### LINKED ISSUE #538: Wire websocket server to the redux store
This issue must be the last one we pick up in the milestone (as that way, it's much easier to implement).

In the context of this issue, we also introduce `lib/typescript/clients/websocket` or `lib/typescript/websocketclient` (depending on what's the naming we chose in #522

### PR DESCRIPTION
**Pull request by bitboxer**
closes #538

### CODE REVIEW

**Review by AitorAlgorta — COMMENTED** [review]
👍

File: lib/typescript/httpclient/index.ts [hunk 1]
```diff
@@ -143,6 +144,13 @@ export class HttpClient {
     }
   }
 
+  public async getConversationInfo(conversationId: string) {
+    const conversation: ConversationPayload = await this.doFetchFromBackend('conversations.info', {
+      conversation_id: conversationId,
```
Comment (AitorAlgorta) [inline lib/typescript/httpclient/index.ts:149]: camelCase?

File: lib/typescript/websocketclient/index.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import {StompWrapper} from './stompWrapper';
+import {Message, Channel} from 'httpclient';
```
Comment (AitorAlgorta) [inline lib/typescript/websocketclient/index.ts:2]: cannot we do all in the same import?

File: lib/typescript/websocketclient/index.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  constructor(token: string, callbackMap: CallbackMap, baseUrl: string) {
+    this.token = token;
+    this.callbackMap = callbackMap;
+    this.apiUrlConfig = `ws://${baseUrl}/ws.communication`;
+
+    this.stompWrapper = new StompWrapper(
+      this.apiUrlConfig,
+      {
+        '/queue/message': item => {
```
Comment (AitorAlgorta) [inline lib/typescript/websocketclient/index.ts:29]: cannot we type the item? honest question

**Review by AitorAlgorta — DISMISSED** [review]
LGTM, small comments

File: lib/typescript/httpclient/index.ts (hunk 1 above)
Comment (bitboxer) [reply lib/typescript/httpclient/index.ts:149]: Not here, this is the actual call to the backend. Or did the backend change to camelCase?

File: lib/typescript/websocketclient/index.ts (hunk 3 above)
Comment (bitboxer) [reply lib/typescript/websocketclient/index.ts:29]: They are typed in the StompWrapper class.

File: lib/typescript/websocketclient/index.ts (hunk 2 above)
Comment (bitboxer) [reply lib/typescript/websocketclient/index.ts:2]: the mappers are not part of the default export or are they?

Comment (AitorAlgorta) [reply lib/typescript/websocketclient/index.ts:2]: they should, I think they are

Comment (AitorAlgorta) [reply lib/typescript/websocketclient/index.ts:2]: but it's ok

File: lib/typescript/websocketclient/index.ts (hunk 3 above)
Comment (AitorAlgorta) [reply lib/typescript/websocketclient/index.ts:29]: ah nice

File: lib/typescript/httpclient/index.ts (hunk 1 above)
Comment (AitorAlgorta) [reply lib/typescript/httpclient/index.ts:149]: ups sorry, you are right

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**